src/router_core/transfer.c

```diff
@@ -1162,7 +1164,7 @@ static void qdr_deliver_continue_CT(qdr_core_t *core, qdr_action_t *action, bool
         }
 
         // This is a multicast delivery
-        if (qdr_is_addr_treatment_multicast(in_dlv->link->owning_addr)) {
+        if (qdr_is_addr_treatment_multicast(in_dlv->link->owning_addr) || in_dlv->multicast || in_dlv->settled) {
```

Comment:
Why did you keep the call to qdr_is_addr_treatment_multicast here?  Isn't it always true that the in_dlv->multicast flag will be initialized appropriately in qdr_link_forward_CT?

Also, why did you add the in_dlv->settled condition?  That is new/changed functionality.

Reply:
If the latter is handling for multi-frame unicasts, then the comment above is no longer accurate.